src/Config.php

```diff
@@ -40,14 +41,24 @@ final class Config
      * @param string $rootPath The path to the project root where composer.json is located.
      * @param string|null $configsPath The path to where configs are stored.
      * @param string|null $environment The environment name.
+     * @param array $mergeRecursive Names of config files that should be merged recursively.
      *
      * @throws ErrorException If the environment does not exist.
      */
-    public function __construct(string $rootPath, string $configsPath = null, string $environment = null)
+    public function __construct(string $rootPath, string $configsPath = null, string $environment = null, array $mergeRecursive = [])
     {
         $this->rootPath = $rootPath;
         $this->relativeConfigsPath = trim($configsPath ?? Options::DEFAULT_CONFIGS_DIRECTORY, '/');
         $this->configsPath = $this->rootPath . '/' . $this->relativeConfigsPath;
+        $this->mergeRecursive = array_merge(
+            $mergeRecursive,
+            [
+                'params.php',
+                'events.php',
+                'events-web.php',
+                'events-console.php'
```

Comment:
That hard-codes some config names. We have `params` already but the rest wasn't forced by yiisoft/config previously. What if I want a config named `events` in my app but don't want to merge it recursively?

Reply:
Default parameters are your idea. In the original implementation, all configs for the recursive merge should be defined in the Config container (ApplicationRunner in the apps templates).

Reply:
He-he. I've changed opinion :)